app/services/password_resetter.go

```diff
@@ -38,5 +40,16 @@ func PasswordResetter(store data.AccountStore, r ops.ErrorReporter, cfg *app.Con
 		return 0, FieldErrors{{"token", ErrInvalidOrExpired}}
 	}
 
+	//Check TOTP MFA
+	if account.TOTPEnabled() {
```

Comment:
How do you feel about requiring TOTP to successfully perform this action?

Originally I was imagining that we could perform the action but wait to establish a session until they verified TOTP, but that might require tackling the concept of a [partially authenticated session](https://github.com/keratin/authn-server/issues/92#issuecomment-481367655).

I think we can reasonably revisit this later without creating a backwards incompatibility. For example, if we figure out partially authenticated sessions then the password reset process could generate either a partial or full session depending on whether the TOTP code was provided.

Reply:
decided here: https://github.com/keratin/authn-server/pull/133#discussion_r342647440